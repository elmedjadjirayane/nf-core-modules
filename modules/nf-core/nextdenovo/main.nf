process NEXTDENOVO {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/fa/fa1c1e961de38d24cf36c424a8f4a9920ddd07b63fdb4cfa51c9e3a593c3c979/data' :
        'community.wave.seqera.io/library/flye:2.9.5--d577924c8416ccd8' }"

    input:
    tuple val(meta), path(reads)
    path config

    output:
    tuple val(meta), path("*.fasta.gz"), emit: fasta
    tuple val(meta), path("*.stat")     , emit: stat
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo ${reads} > input.fofn
    nextDenovo \\
        $config \\
        input.fofn \\

    gzip -c ./03.ctg_graph/nd.asm.fasta > ${prefix}.assembly.fasta.gz
  
    mv ./03.ctg_graph/nd.asm.fasta.stat ${prefix}.assembly_info.stat

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        \$( nextDenovo --version )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo stub | gzip -c > ${prefix}.assembly.fasta.gz
    echo contig_1 > ${prefix}.assembly_info.stat

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
    \$( nextDenovo --version )
    END_VERSIONS
    """
}
