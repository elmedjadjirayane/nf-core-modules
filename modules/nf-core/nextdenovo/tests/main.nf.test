// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test nextdenovo
nextflow_process {

    name "Test Process NEXTDENOVO"
    script "../main.nf"
    process "NEXTDENOVO"

    tag "modules"
    tag "modules_nfcore"
    tag "nextdenovo"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("nextdenovo_ont") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/sarscov2/nanopore/fastq/test.fastq.gz', checkIfExists: true),
                ]
                input[1] = file(params.modules_testdata_base_path + 'generic/txt/hello.txt')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    

}
