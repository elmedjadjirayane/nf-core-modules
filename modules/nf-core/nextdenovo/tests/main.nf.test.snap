{
    "nextdenovo_ont": {
        "content": [
            "798439cbd7fd81cbcc5078022dc5479d",
            [
                "versions.yml:md5,ed7a2cfe2387b964c06b6c6847f46996"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T12:42:15.405989"
    }
   
}